apps/api/src/domain/models/tag/tag.entity.ts

```diff
@@ -1,17 +1,41 @@
+import { InternalError } from '../../../lib/errors/InternalError';
+import cloneToPlainObject from '../../../lib/utilities/cloneToPlainObject';
+import { DeepPartial } from '../../../types/DeepPartial';
 import { DTO } from '../../../types/DTO';
+import { Valid } from '../../domainModelValidators/Valid';
+import { HasEntityIdAndLabel } from '../../interfaces/HasEntityIdAndLabel';
+import { ValidatesItsExternalState } from '../../interfaces/ValidatesItsExternalState';
 import { EntityId } from '../../types/ResourceId';
+import { InMemorySnapshot } from '../../types/resourceTypes';
+import validateReferencesAgainstExternalState from '../../utilities/validation/validateReferencesAgainstExternalState';
 import BaseDomainModel from '../BaseDomainModel';
+import InvalidExternalReferenceInCategoryError from '../categories/errors/InvalidExternalReferenceInCategoryError';
+import { ResourceOrNoteCompositeIdentifier } from '../categories/types/ResourceOrNoteCompositeIdentifier';
 
-export class Tag extends BaseDomainModel {
+export class Tag extends BaseDomainModel implements ValidatesItsExternalState, HasEntityIdAndLabel {
     id: EntityId;
 
     label: string;
 
-    constructor(dto: DTO<Tag>) {
+    members: ResourceOrNoteCompositeIdentifier[];
+
+    constructor({ id, label, members }: DTO<Tag>) {
         super();
 
-        this.id = dto.id;
+        this.id = id;
+
+        this.label = label;
+
+        this.members = cloneToPlainObject(members);
+    }
 
-        this.label = dto.label;
+    // TODO Use a mixin for this
+    validateExternalState(externalState: DeepPartial<InMemorySnapshot>): Valid | InternalError {
+        return validateReferencesAgainstExternalState(
```

Comment:
Just wondering about the word `References`, could that be made more specific?

Reply:
done